src/dialog-wrapper.component.ts

```diff
@@ -51,6 +56,26 @@ export class DialogWrapperComponent {
     this.content.wrapper = this.wrapper;
     return this.content;
   }
+
+  onClickOutsideModalContent( callback: () => void) {
+    this.clickOutsideCallback = callback;
+    const containerEl = this.wrapper.nativeElement;
+    containerEl.querySelector(':first-child').addEventListener('click', this.stopEventPropagation);
```

Comment:
Removed the implicit dependency on bootstraps modal-content class here. Really the point is simply, click the background - close, click any of my children e.g. modal  stuff, do not close.